code/game/machinery/_machinery.dm

```diff
@@ -542,3 +548,48 @@ Class Procs:
 	. = . % 9
 	AM.pixel_x = -8 + ((.%3)*8)
 	AM.pixel_y = -8 + (round( . / 3)*8)
+
+/obj/machinery/proc/id_insert(mob/user, obj/item/card/id/I, obj/item/card/id/target_id)
+	if(target_id)
+		to_chat(user, "<span class='warning'>There's already an ID card in the console!</span>")
+		return
+	if(istype(I))
+		if(!user.transferItemToLoc(I, src))
+			return
+		target_id = I
+		user.visible_message("<span class='notice'>[user] inserts an ID card into the console.</span>", \
+							"<span class='notice'>You insert the ID card into the console.</span>")
+		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+	else
+		to_chat(user, "<span class='danger'>No valid ID.</span>")
+	updateUsrDialog()
+
+/obj/machinery/proc/id_eject(mob/user, obj/item/card/id/I, obj/item/card/id/target_id)
```

Comment:
you don't need the second parameter here, you're not using it in the proc.

Reply:
Thanks, I got it to compile, but it didn't work though...